api/apps/graphql/index.coffee

```diff
@@ -0,0 +1,34 @@
+express = require 'express'
+graphqlHTTP = require 'express-graphql'
+joiql = require 'joiql'
+{ string, object, array } = require 'joi'
+Article = require '../articles/model/schema'
+{ find, where, presentCollection } = require '../articles/model'
+
+app = module.exports = express()
+
+api = joiql
+  query:
+    articles: array().items(object(
+      Article.inputSchema
+    )).meta(
+      args: Article.querySchema
+    )
+
+api.use (ctx, next) ->
```

Comment:
A suggestion for testing these parts—it would be good to extract these functions into testable libraries. A "resolvers" layer is a popular abstraction in GraphQL, so you could start organizing things that way e.g. this woudl be `api.use resolvers.articles` and this middleware function would be moved out to resolvers/articles.coffee, and exported e.g.

````coffeescript
module.exports.articles = (ctx, next) ->
````

You could then write require them into tests and writing tests similar to how we test routing middleware functions—by stubbing the `ctx` and `next` objects/functions passed in.